modules/text_server_adv/text_server_adv.cpp

```diff
@@ -6145,13 +6145,19 @@ _FORCE_INLINE_ void TextServerAdvanced::_add_featuers(const Dictionary &p_source
 	}
 }
 
-UBreakIterator *TextServerAdvanced::ShapedTextDataAdvanced::_get_break_iterator_for_locale(const String &p_language, UErrorCode *r_err) {
-	HashMap<String, UBreakIterator *>::Iterator key_value = line_break_iterators_per_language.find(p_language);
+UBreakIterator *TextServerAdvanced::_create_break_iterator_for_locale(const String &p_language, UErrorCode *r_err) const {
+	// Creating UBreakIterator (ubrk_open) is surprisingly costly.
+	// However, cloning (ubrk_clone) is cheaper, so we keep around blueprints to accelerate creating new ones.
+
+	const String language = p_language.is_empty() ? TranslationServer::get_singleton()->get_tool_locale() : p_language;
+	_THREAD_SAFE_METHOD_
+	const HashMap<String, UBreakIterator *>::Iterator key_value = line_break_iterators_per_language.find(language);
 	if (key_value) {
-		return key_value->value;
+		return ubrk_clone(key_value->value, r_err);
 	}
-	UBreakIterator *brk = ubrk_open(UBRK_LINE, p_language.is_empty() ? TranslationServer::get_singleton()->get_tool_locale().ascii().get_data() : p_language.ascii().get_data(), nullptr, 0, r_err);
-	return line_break_iterators_per_language.insert(p_language, brk)->value;
+	UBreakIterator *brk = ubrk_open(UBRK_LINE, language.ascii().get_data(), nullptr, 0, r_err);
```

Comment:
`ubrk_open` can fail if engine is built without ICU database (export templates always are by default), so this should check for `r_err` before trying to clone it.

Reply:
True. It's somewhat questionable what to do then, but I opted not to cache the failure and merely return. This matches previous behaviour and sounds sane. Let me know if you'd prefer something else.